src/file_system/mod.rs

```diff
@@ -397,6 +399,14 @@ impl DirectoryContents {
     pub fn file(filename: Label, contents: &[u8]) -> Self {
         DirectoryContents::File(File::new(filename, contents))
     }
+
+    pub fn label(&self) -> Option<&Label> {
+        match self {
+            DirectoryContents::SubDirectory(dir) => Some(&dir.label),
+            DirectoryContents::File(file) => Some(&file.filename),
+            _ => None
```

Comment:
Explicit match here, please. In the case of the `enum` getting extended we'll get a compiler warning

Reply:
Yup, makes total sense